### PR TITLE
Add unit tests to cli/command/formatter/stats.go

### PR DESCRIPTION
**Pull request by boaz0**
**\- What I did**
Add unit tests to `ContainerStatsContext` and `ContainerStatsContextWrite`.

**\- How I did it**
- Created cli/command/formatter/stats_test.go
- Add unit tests to cli/command/formatter/stats.go

**\- How to verify it**
Run `cd cli/command/formatter && go test -coverprofile=cover.out && go tool cover -func=cover.out | grep stats.go`

**\- Description for the changelog**

**\- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: cli/command/formatter/stats_test.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+		{StatsEntry{PidsCurrent: 10}, "10", pidsHeader, ctx.PIDs},
+		{StatsEntry{PidsCurrent: 10, IsInvalid: true}, "--", pidsHeader, ctx.PIDs},
+		{StatsEntry{PidsCurrent: 10, OSType: "windows"}, "--", pidsHeader, ctx.PIDs},
+	}
+
+	for _, te := range tt {
+		ctx = containerStatsContext{s: te.stats}
+		v := te.call()
+		if v != te.expValue {
+			t.Fatalf("Expected %s, was %s\n", te.expValue, v)
```
Comment (AkihiroSuda) [inline cli/command/formatter/stats_test.go]: you don't need `\n` in `t.Fatalf`

File: cli/command/formatter/stats_test.go [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+				NetworkRx:        30,
+				NetworkTx:        30,
+				BlockRead:        30,
+				BlockWrite:       30,
+				PidsCurrent:      3,
+				IsInvalid:        true,
+				OSType:           "linux",
+			},
+		}
+		out := bytes.NewBufferString("")
```
Comment (AkihiroSuda) [inline cli/command/formatter/stats_test.go]: nit: `var out bytes.Buffer`  (You don't need to use `NewBufferString` here)
https://golang.org/pkg/bytes/#Buffer

File: cli/command/formatter/stats_test.go (hunk 1 above)
Comment (ripcurld00d) [inline cli/command/formatter/stats_test.go]: :+1:

File: cli/command/formatter/stats_test.go (hunk 2 above)
Comment (ripcurld00d) [inline cli/command/formatter/stats_test.go]: :+1:

**Comment by ripcurld00d** [comment]
@AkihiroSuda thanks for your review.
I will update the commit in the next hours.
:coffee: :grinning:

**Comment by ripcurld00d** [comment]
Need to rebase to #27759 and update the tests accordingly. :shell:

**Comment by ripcurld00d** [comment]
OK, @AkihiroSuda @thaJeztah @jhowardmsft your feedback please? :sake: 

Thanks

File: cli/command/formatter/stats.go [hunk 3]
```diff
@@ -167,7 +167,7 @@ func (c *containerStatsContext) MemUsage() string {
 func (c *containerStatsContext) MemPerc() string {
 	header := memPercHeader
 	c.AddHeader(header)
-	if c.s.IsInvalid {
+	if c.s.IsInvalid || c.s.OSType == winOSType {
```
Comment (ripcurld00d) [inline cli/command/formatter/stats.go]: @jhowardmsft, I added this part because I thought Windows doesn't collect memory percentage stats, is that right?

Comment (jhowardmsft) [inline cli/command/formatter/stats.go]: Correct

Comment (ripcurld00d) [inline cli/command/formatter/stats.go]: Thanks :+1:

File: cli/command/formatter/stats_test.go [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+				NetworkRx:        30,
+				NetworkTx:        30,
+				BlockRead:        30,
+				BlockWrite:       30,
+				PidsCurrent:      3,
+				IsInvalid:        true,
+				OSType:           "windows",
+			},
+		}
+		out := bytes.NewBufferString("")
```
Comment (AkihiroSuda) [inline cli/command/formatter/stats_test.go]: var out bytes.Buffer

File: cli/command/formatter/stats_test.go [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+		{StatsEntry{Memory: 24, MemoryLimit: 30, OSType: "windows"}, "24 B", winMemUseHeader, ctx.MemUsage},
+		{StatsEntry{PidsCurrent: 10}, "10", pidsHeader, ctx.PIDs},
+		{StatsEntry{PidsCurrent: 10, IsInvalid: true}, "--", pidsHeader, ctx.PIDs},
+		{StatsEntry{PidsCurrent: 10, OSType: "windows"}, "--", pidsHeader, ctx.PIDs},
+	}
+
+	for _, te := range tt {
+		ctx = containerStatsContext{s: te.stats}
+		v := te.call()
+		if v != te.expValue {
```
Comment (AkihiroSuda) [inline cli/command/formatter/stats_test.go]: if v := .. ; v != .. { 
  .. 
}

Comment (ripcurld00d) [inline cli/command/formatter/stats_test.go]: :+1:

File: cli/command/formatter/stats_test.go (hunk 4 above)
Comment (ripcurld00d) [inline cli/command/formatter/stats_test.go]: :+1: Thanks!

**Review by vdemeester — APPROVED** [review]
LGTM 🐸

**Review by thaJeztah — APPROVED** [review]
thanks for helping review @AkihiroSuda 

LGTM

**Pull request merged**